source/JsMaterialX/JsMaterialXCore/JsValue.cpp

```diff
@@ -0,0 +1,49 @@
+#include "../helpers.h"
+#include <MaterialXCore/Value.h>
+
+#include <emscripten.h>
+#include <emscripten/bind.h>
+
+using namespace emscripten;
+
+namespace mx = MaterialX;
+
+using namespace mx;
+
+#define BIND_TYPE_INSTANCE(NAME, T)                                                   \
+    class_<TypedValue<T>, base<Value>>("TypedValue_" #NAME) \
+        .smart_ptr<std::shared_ptr<TypedValue<T>>>("TypedValue<T>")                         \
+        .function("getData", &TypedValue<T>::getData)                                 \
+        .function("getValueString", &TypedValue<T>::getValueString)                   \
+        .class_function("createValue", &Value::createValue<T>)                        \
+        .class_property("TYPE", &TypedValue<T>::TYPE);
+
+extern "C"
+{
+    EMSCRIPTEN_BINDINGS(value)
+    {
+        class_<Value>("Value")
+            .smart_ptr<std::shared_ptr<Value>>("Value")
+            .smart_ptr<std::shared_ptr<const Value>>("Value")
+            .function("getValueString", &Value::getValueString)
+            .function("getTypeString", &Value::getTypeString)
+            .class_function("createValueFromStrings", &Value::createValueFromStrings);
+
+        BIND_TYPE_INSTANCE(integer, int)
```

Comment:
There are compiler warnings here like this:
```
D:/Work/materialx/MaterialX-adsk-dev/source/JsMaterialX/..\MaterialXCore/Value.h:183:25: note: forward declaration of template entity is here
    static const string TYPE;
                        ^
../JsMaterialXCore/JsValue.cpp:47:9: note: add an explicit instantiation declaration to suppress this warning if 'MaterialX::TypedValue<std::__2::vector<std::__2::basic_string<char>, std::__2::allocator<std::__2::basic_string<char>>>>::TYPE' is explicitly instantiated in another translation unit
        BIND_TYPE_INSTANCE(stringarray, StringVec)
        ^
../JsMaterialXCore/JsValue.cpp:19:49: note: expanded from macro 'BIND_TYPE_INSTANCE'
        .class_property("TYPE", &TypedValue<T>::TYPE);
```

Reply:
This was addressed in commit: 6a08be7ea8c666f1c24d088f6f7a6317fa1b95a0